### DISCUSSION TITLE
Does VS code auto update Python dependencies in virtual environment?

### DISCUSSION THREAD

**Discussion by anguslou** [microsoft/vscode-discussions#307] [Extension Development QnA] [answered]
I am new to VS code. Recently I code in VS code for a python project and run by conda virtual environment. Later I found some dependencies were updated. Does VS code auto-update dependencies in virtual environment when it runs the code? How to stop it if yes?

**Comment by alefragnani** [comment, marked as answer]
Hi @anguslou ,

I'm not a Python developer nor have any Python extension installed, but I suppose that if this is happening, it is something inherited from the Python extension that you have installed.

A quick look at the [Microsoft Python Extension](https://marketplace.visualstudio.com/items?itemName=ms-python.python), it doesn't seems to have any setting for that, just a `python.condaPath`, which is a _Path to the conda executable to use for activation_. If that's the extension you are using, there is a good probability that this is the _default behavior_

I suggest you to review the settings available in the Python extension you are using, and if no success, ask the same question in the repository of that extension.

Hope this helps

**Reply by isidorn** [reply]
In addition to this great answer, let me add that VS Code does not auto-update dependencies in virtual envs.
